git/object_scanner.go

```diff
@@ -202,36 +143,34 @@ func IsMissingObject(err error) bool {
 	return ok
 }
 
-// scan scans for and populates a new Git object given an OID.
 func (s *ObjectScanner) scan(oid string) (*object, error) {
-	if _, err := fmt.Fprintln(s.to, oid); err != nil {
-		return nil, err
-	}
-
-	l, err := s.from.ReadBytes('\n')
+	var (
+		obj      gitobj.Object
+		size     int64
+		contents *io.LimitedReader
+	)
+
+	oidbytes, _ := hex.DecodeString(oid)
```

Comment:
Nice, I think that it makes sense to ignore this error, since it'll turn into a `&missingErr` shortly after anyway.

Since we throw this value away, I wonder if it makes sense to introduce a `Must*` version of this function, a-la `regexp.Compile` and `regexp.MustCompile`. Maybe:

```go
func mustDecode(s string) []byte {
        x, _ := hex.DecodeString(s)
        return x
}
```

which we could then use directly as:

```go
obj, err := s.gitobj.Object(mustDecode(oid))
```

This isn't _quite_ true to the spirit of `regexp.MustCompile`'s pattern, since the standard library's implementation `panic()`s if `err != nil`, but I think that this is close enough.

Alternatively, maybe we should introduce `ObjectString`, `BlobString`, (etc.) onto `gitobj`. What do you think?

Reply:
I think a mustDecode function is a good idea here.